src/VisualStudio/Core/Test/Options/BasicEditorConfigGeneratorTests.vb

```diff
@@ -73,11 +74,18 @@ dotnet_style_prefer_is_null_check_over_reference_equality_method = true:suggesti
 # Field preferences
 dotnet_style_readonly_field = true:suggestion
 
+# Parameter preferences
+dotnet_style_unused_parameters = all:suggestion
```

Comment:
calling this 'style' is a mistake to em.  Note: this doesn't apply to the 'expr level' features.  In that case, it really could be a style as it's an entirely internal method choice that you have control over and can decide how you want this to be controlled.


'unused/used' parameters though is not any sort of 'style'.  no one says "my style is to not have these".

Reply:
This was presented at the design meeting and it was decided to keep "all" as the default, given the diagnostic message also indicates whenever recommendation is for a public API.

---
In reply to: [230606115](https://github.com/dotnet/roslyn/pull/30777#discussion_r230606115) [](ancestors = 230606115)

Reply:
Do you have a better name suggestion? A different name would likely mean this will need to go under a separate new tab in Tools Options. Tagging @sharwell @jinujoseph 

---
In reply to: [230606172](https://github.com/dotnet/roslyn/pull/30777#discussion_r230606172) [](ancestors = 230606172)

Reply:
in terms of 'all', that decisoin is fine with me.  thanks for checking on it.

in terms of a naming i think "code_quality" instead of "code_style".  It makes sense as a different type of bucket for me, and itself would have subcategories for areas/apis potentially.

Reply:
This does seem more like "code quality" rather than "code style". Somebody suggested previously that there should really be two tabs in Tools Options - code style and code quality, both of which would have the same UI / same presentation, with different analyzers.

Reply:
> Somebody suggested previously that there should really be two tabs in Tools Options - code style and code quality, both of which would have the same UI / same presentation, with different analyzers.

That was me :-), but I am too lazy to do the work to fork the UI. Regardless, I am going to change the editor config option name to have "quality" rather then "style", but retain the option in the existing Code Style page for this PR (which is already becoming too big) and file a separate issue to add a new Code Quality tab in the UI and move this option there.

Reply:
Filed [https://github.com/dotnet/roslyn/issues/30986](https://github.com/dotnet/roslyn/issues/30986) for the UI work.

---
In reply to: [231171949](https://github.com/dotnet/roslyn/pull/30777#discussion_r231171949) [](ancestors = 231171949)